README.md

```diff
@@ -112,6 +112,136 @@ The script has been tested with the KALDI Git hash "7637de77e0a77bf280bef9bf484e
+
+On the file system you will find files like `.../observation/train_si284/0_4axc0218_01kc020f.wav`.
+Here an explanation, how the path and file name are generated:
+-  `observation`: signal type
+    -  `observation` = `early` + `tail` + `noise`
+    -  `speech_source` convolved with `rirs` = `early` + `tail`
+    -  `speech_source`: The padded signal from WSJ (i.e. `original_source`).
+    -  `early`/`tail`: `speech_source` convolved with inital/late part of `rirs`
+-  `train_si284`/`cv_dev93`/`test_eval92`: The original WSJ dataset name.
+-  `0_4axc0218_01kc020f`:
+    -  The first part is a running index for the generated `rirs`
+    -  The second and third part are the speaker ids
+    -  An optional fourth part is added, when the file is speaker specific, e.g. `speech_source`, `early` and `tail`
+
+The database creation generates a json file. This file contains all information about the database.
+The pattern is as follows:
+```python
+{
+  "datasets": {
+    dataset_name: {  # "train_si284", "cv_dev93" or "test_eval92"
+      example_id: {  # <rir_index>_<first_speaker_id>_<second_speaker_id>
+        "room_dimensions": [[...], [...], [...]]],
+        "sound_decay_time": ...,
+        "source_position": [[..., ...], [..., ...], [..., ...]]
+        "sensor_position": [[..., ..., ..., ..., ..., ...], [..., ..., ..., ..., ..., ...], [..., ..., ..., ..., ..., ...]],
+        "example_id": "...",
+        "num_speakers": 2,
+        "speaker_id": ["...", "..."],
+        "gender": ["...", "..."],  # "male" or "female"
+        "kaldi_transcription": ["...", "..."],
+        "log_weights": [..., ...],  # weights of utterances before the are added
+        "num_samples": {
+          "original_source": [..., ...],
+          "observation": ...,
+        },
+        "offset": [..., ...]  # Offset of utterance start in samples
+        "snr": ...,
+        "audio_path": {
+          "original_source": ["...", "..."],
+          "speech_source": ["...", "..."],
+          "rir": ["...", "..."],
+          "speech_reverberation_early": ["...", "..."],
+          "speech_reverberation_tail": ["...", "..."],
+          "noise_image": "...",
+          "observation": "...",
+        }
+      }
+    }
+  }
+}
+```
+This file can be used to get all details for each example.
+To read it with python, we have some helper functions:
+<!-- To access one (or multiple) examples, you can use the following setup code: -->
+
+```python
+from sms_wsj.database import SmsWsj, AudioReader
+db = SmsWsj(json_path='.../sms_wsj.json')
+ds = db.get_dataset('train_si284')  # "train_si284", "cv_dev93" or "test_eval92"
+ds = ds.map(AudioReader((
+    'observation',
+    'speech_source',
+    # 'original_source',
+    # 'speech_reverberation_early',
+    # 'speech_reverberation_tail',
+    'speech_image',
+    # 'noise_image',
+    # 'rir',
+)))
+ds = ds.shuffle(reshuffle=True)
```

Comment:
Why do you need shuffle here?

Reply:
I think removing or comment code is much easier, than figuring out, how to correctly prepare the data. Especially for the training data IMHO the shuffle is important.

Reply:
Sure but this is not a neural network tutorial but an introduction to the database and in that context the shuffle is not necessary

Reply:
Ok, I commented it. But I think it is still useful for a reader to get a hint, how he/she can perform the most common option.

(Btw. it would also be some advertisement for lazy_dataset)